src/paginationCalls/pageDescriptor.ts

```diff
@@ -103,6 +116,108 @@ export class PageDescriptor implements Descriptor {
     return stream;
   }
 
+  /**
+   * Create an async iterable which can be recursively called for data on-demand.
+   */
+  asyncIterate(
+    apiCall: GaxCall,
+    request: RequestType,
+    options: CallSettings
+  ): AsyncIterable<{} | undefined> {
+    const iterable = this.createIterator(options);
+    const funcPromise =
+      typeof apiCall === 'function' ? Promise.resolve(apiCall) : apiCall;
+    funcPromise
+      .then((func: GaxCall) => {
+        this.resolveParams(request, func, options);
+      })
+      .catch(error => {
+        throw new Error(error);
+      });
+    return iterable;
+  }
+
+  createIterator(options: CallSettings): AsyncIterable<{} | undefined> {
+    const requestPageTokenFieldName = this.requestPageTokenField;
+    const responsePageTokenFieldName = this.responsePageTokenField;
+    const resourceField = this.resourceField;
+    const self = this;
+    const getNextPageRequest = this.getNextPageRequest;
```

Comment:
It might not work as you expect. Maybe
```ts
const getNextPageRequest = (...args) => { this.getNextPageRequest(...args); } // make a closure, capture "this"
```
will do what you expect (just check if `this` inside `getNextPageRequest` points to what you need)

Reply:
Yes, I was confused by this problem, I tried to use `this` in `getNextPageRequest` method, but it does not refer to what I have expected.

Reply:
So I just passed whatever I need as const to the method like `resourceField, requestPageTokenField ......`